### DISCUSSION TITLE
Ability to pre-populate/set value of ui.upload

### DISCUSSION THREAD

**Discussion by AdamC03** [zauberzeug/nicegui#2647] [Ideas / Feature Requests]
The ability to set a "value" to the ui.upload component will be extremely useful and it is inherently possible within the qupload component as shown here:
![image](https://github.com/zauberzeug/nicegui/assets/128116641/9363e218-dd3f-4b01-9f8c-bbf873aa9730)
![image](https://github.com/zauberzeug/nicegui/assets/128116641/52c5b867-bb4f-415b-968f-fb6713517cb3)

One example of how its useful is as follows:
Modifying a document, say there is a form that you can fill out and a backed generates a pdf but you later would like to make changes to the pdf by allowing the form that created it to be edited. All other data entires can have their values pre-populated by setting their values property. This can't be done with ui.upload, ideally you could pre-populate this with images that were submitted so these too can be edited without all data needing to be re-attached.

**Comment by falkoschindler** [comment]
Hi @DavidLeBonk,

I'm not quite sure if I get the idea.
Apart from that, "files" is a computed prop and I think we can't set it programmatically.
And where is the second screenshot coming from? What's its purpose?

**Reply by AdamC03** [reply]
The purpose is to have a ui.upload be already populated on creation. The second screenshot is to display a suggestion for how it can be done (via dynamically creating files in form data and attaching them to the upload component)

**Reply by frettig-radiox** [reply]
Hi! 

I came across this post and I'm facing the same issue: I'd like to have ui.upload pre-populated with previously uploaded files when the page loads.

Would really appreciate any insights or updates. Thanks so much!
